nautobot/core/settings_funcs.py

```diff
@@ -82,3 +84,93 @@ def parse_redis_connection(redis_database):
+    root_level="INFO",
+    debug_db=False,
+    plain_format=False,
+) -> None:
+    """Set up structlog logging for Django."""
+    django_logging["version"] = 1
+    django_logging["disable_existing_loggers"] = True
+    if "test" in sys.argv:
+        django_logging["handlers"] = {
+            "null_handler": {
+                "level": "INFO",
+                "class": "logging.NullHandler",
+            },
+        }
+        for logger in django_logging["loggers"].values():
+            logger["handlers"] = ["null_handler"]
+            logger["level"] = "INFO"
+
+        return
+
+    django_apps.append("django_structlog")
+    django_middleware.append("django_structlog.middlewares.RequestMiddleware")
+
+    processors = (
+        # Add the log level to the event dict under the level key.
+        structlog.stdlib.add_log_level,
+        # Add the logger name to the event dict.
+        structlog.stdlib.add_logger_name,
+        # Apply stdlib-like string formatting to the event key.
+        structlog.stdlib.PositionalArgumentsFormatter(),
+        # A processor that merges in a global (context-local) context.
+        structlog.contextvars.merge_contextvars,
+        # Add a timestamp to event_dict in ISO 8601 format.
+        structlog.processors.TimeStamper(fmt="iso"),
+        # Replace an exc_info field with an exception string field using Python's built-in traceback formatting.
+        structlog.processors.format_exc_info,
+        # Add stack information with key stack if stack_info is True.
+        structlog.processors.StackInfoRenderer(),
+        # Decode byte string values in event_dict.
+        structlog.processors.UnicodeDecoder(),
+    )
+
+    django_logging["formatters"] = {
+        "default_formatter": {
+            "()": structlog.stdlib.ProcessorFormatter,
+            "foreign_pre_chain": processors,
+            "processor": structlog.dev.ConsoleRenderer() if plain_format else structlog.processors.JSONRenderer(),
+        },
+    }
+
+    django_logging["handlers"] = {
+        "default_handler": {
+            "level": "DEBUG",
+            "class": "logging.StreamHandler",
+            "formatter": "default_formatter",
+        },
+    }
+
+    django_logging["root"] = {
+        "handlers": ["default_handler"],
+        "level": root_level,
+    }
+
+    if debug_db:
+        django_logging["loggers"]["django.db.backends"] = {"level": "DEBUG"}
+
+    for logger in django_logging["loggers"].values():
+        if "level" not in logger:
+            logger["level"] = log_level
+        logger["propagate"] = False
+        logger["handlers"] = ["default_handler"]
+
+    structlog.configure(
+        processors=[
+            # Filter out log records below the configured log level.
+            structlog.stdlib.filter_by_level,
+            *processors,
+            # Must be the final processor in structlog's processor chain.
+            structlog.stdlib.ProcessorFormatter.wrap_for_formatter,
```

Comment:
Why aren't `filter_by_level` and `wrap_for_formatter` just declared as part of the `processors` list itself?

Reply:
`processors` are used in `default_formatter` as well, and these are necessary only withing this configuration, but not within `default_formatter`